src/Compilers/CSharp/Portable/Symbols/SymbolWithAnnotations.cs

```diff
@@ -267,11 +283,16 @@ public bool Equals(TypeSymbolWithAnnotations other, TypeCompareKind comparison)
                 return false;
             }
 
-            if ((comparison & TypeCompareKind.CompareNullableModifiersForReferenceTypes) != 0 && other.IsNullable != this.IsNullable)
+            if ((comparison & TypeCompareKind.CompareNullableModifiersForReferenceTypes) != 0)
```

Comment:
Nit, not related to this PR: we may want to think about renaming the compare options. Maybe `ConsiderNullableAnnotations` and `ConsiderNonNullTypes` (or maybe `Ignore...`).

Reply:
Added `// PROTOTYPE` comment.

---
In reply to: [203851846](https://github.com/dotnet/roslyn/pull/28687#discussion_r203851846) [](ancestors = 203851846)